packages/mrgn-ui/src/components/wallet-v2/components/wallet-auth-accounts/wallet-auth-accounts.tsx

```diff
@@ -294,7 +269,7 @@ export const WalletAuthAccounts = ({
               <div className={cn("grid gap-2", isActivatingAccount !== null && "pointer-events-none animate-pulsate")}>
                 {marginfiAccounts.map((account, index) => {
                   const isActiveAccount = selectedAccount && selectedAccount.address.equals(account.address);
-                  const accountLabel = accountLabels[account.address.toBase58()] || `Account ${index + 1}`;
+                  const accountLabel = accountLabels?.[account.address.toBase58()] || `Account ${index + 1}`;
```

Comment:
the conditional above ensures accountLabels?.[selectedAccount.address.toBase58()] is defined before entering this part of the code. Is this an potential issue since it will never reach `Account ${index + 1}` ?

Reply:
Good catch! Didnt think of the conditional render of the button above. It should never get to a state where the acccountLabels, so I dont think this is an issue. We could leave out the `Account ${index + 1}` and just add in an `!`, but I dont think it'll make any difference.

Reply:
aah I see know, your answer had me a bit confused, as the conditional should never have any child complain about potentially being undefined

but I just noticed that the conditional is for "selectedAccount"  and not the accounts you're mapping

Reply:
so your current code is structured correctly right?

Reply:
Yes, also got confused myself lol

Like you said, the conditional rendering of the popover button is for the selectedAccount itself, not for all accounts. So yes, it should try to read accountLabel for a given address, if not it should display `Account X`.